### PR TITLE
Add benchmarks for rescan components

### PR DESCRIPTION
**Pull request by str4d**
Part of #1749.

### CODE REVIEW

File: src/utiltest.cpp [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+
+    // Empty output script.
+    CScript scriptCode;
+    CTransaction signTx(mtx);
+    uint256 dataToBeSigned = SignatureHash(scriptCode, signTx, NOT_AN_INPUT, SIGHASH_ALL);
+
+    // Add the signature
+    assert(crypto_sign_detached(&mtx.joinSplitSig[0], NULL,
+                         dataToBeSigned.begin(), 32,
+                         joinSplitPrivKey
```
Comment (daira) [inline src/utiltest.cpp]: Nitpick: indentation

**Comment by daira** [comment]
ut(ACK+cov)

**Comment by str4d** [comment]
@bitcartel you run the benchmarks using `performance-benchmarks.sh`, the same as all the others (and in particular, the same way as `verifyjoinsplit` which also requires a `JoinSplit`):

- `./qa/zcash/performance-measurements.sh time trydecryptnotes` - tries to decrypt the supplied `JoinSplit` with 10 random z-addrs.
  - Could make the 10 a parameter if desired.
  - Just realised that as we don't need valid proofs, we could remove the JS argument and instead generate them on-the-fly like for `incnotewitnesses`.
- `./qa/zcash/performance-measurements.sh time incnotewitnesses NUM_TXS` - increments the witnesses of a transaction
  - Just spotted an issue: there are no JSs in the second block, so part of the incrementing code isn't being exercised.

**Comment by bitcartel** [comment]
Ok, I was using RPC calls directly.  I don't think how to run the benchmarks is documented anywhere for users.
Typo above: "performance-benchmarks.sh" -> "performance-measurements.sh" ;-)

**Comment by str4d** [comment]
Self-ACK (I have addressed my comments).

**Comment by str4d** [comment]
I'll set this merging in an hour or so at the latest if I don't see further ACKs, since this is only benchmarking code and most of it has already been ACKed once.

**Comment by bitcartel** [comment]
Testing now.

**Comment by str4d** [comment]
@zkbot r+

**Pull request merged**